packages/visual-editor/src/components/Dropzone/useComponent.ts

```diff
@@ -69,31 +71,48 @@ export const useComponent = ({
     areEntitiesFetched,
     resolveDesignValue,
     renderDropzone,
-    definition: componentRegistration.definition,
+    definition: componentRegistration?.definition,
     userIsDragging,
     slotId,
   });
 
   // Only pass editor props to built-in components
   const { editorMode, renderDropzone: _renderDropzone, ...otherComponentProps } = componentProps;
-  const elementToRender = builtInComponents.includes(node.data.blockId || '')
-    ? (dragProps?: NoWrapDraggableProps) =>
-        React.createElement(componentRegistration.component, { ...dragProps, ...componentProps })
-    : node.type === ASSEMBLY_NODE_TYPE
-      ? // Assembly.tsx requires renderDropzone and editorMode as well
-        () => React.createElement(componentRegistration.component, componentProps)
-      : () =>
-          React.createElement(
-            ImportedComponentErrorBoundary,
-            null,
-            React.createElement(componentRegistration.component, otherComponentProps),
-          );
+  const createElementToRender = (componentRegistration: ComponentRegistration) => {
```

Comment:
I like this update, the code is easier to read now. 👏

Reply:
Or was it? I think I spoke about L97.. but you spoke about L81... :)

Reply:
✅  RESOLVEd